### DISCUSSION TITLE
want to bind form values into another component on every field change

### DISCUSSION THREAD

**Discussion by Mohamedsobhy292** [react-hook-form/react-hook-form#1867] [Q&A] [answered]
I have a react-hook-form which I'm using controlled fields on it , and I 've another component that needs to display the values of the form on every change 

what I'm trying to do is, I watch all the values and  on every change, I call a method to update the other component state

```
const Form = () => {
    const methods = useForm()
    const { handleSubmit, watch } = methods
    const { updateUserData } = useContext(AppContext)
    const formValues = watch()

    useEffect(() => {
        updateUserData(formValues)
    }, [formValues])
```

the problem here is that form values is an object and it will trigger rerender and I will have infinite loop with `Maximum update depth exceeded` 

is there any way to do this instead of having to list every single form value on the dependency array  ?

**Comment by bluebill1049** [comment, marked as answer]
`validate` with `getValues()`?

**Reply by Mohamedsobhy292** [reply]
@bluebill1049 
can you explain please ?

**Reply by bluebill1049** [reply]
```
register({
  validate: value => getValues('anotherFIeld') === value
})
```

**Reply by Mohamedsobhy292** [reply]
I think i'm not explaining it right, I'm using the controlled input so can't use the register propety, I want to get the values on every change on this form inputs to display it on another component 
here's some similar example https://codesandbox.io/s/pensive-galileo-fxh4s?file=/src/App.js ,

what I'm trying to do this when the user type anything on the inputs it get displayed on the other component , I commented `setFormValues` on the example because when I run it i've a problem of `Maximum update depth exceeded`

**Reply by bluebill1049** [reply]
why not use `Controller`? it take care of the register for you with Controlled input.

**Reply by Mohamedsobhy292** [reply]
yeah I'm already doing that, what I want to do is when any value on the form changes , the change is viewed on the other component immediately with no need to click submit

**Reply by bluebill1049** [reply]
`watch`?

**Reply by Mohamedsobhy292** [reply]
yeah I'm using `watch()` to watch all values and set to the other component, the problem, as I mentioned, is that when adding `values` to useffect `deps array` I got infinite loop with Maximum update depth exceeded

**Reply by bluebill1049** [reply]
why do you need `useEffect`?  why not let it flow before render?

**Reply by Mohamedsobhy292** [reply]
because I want to get the form values on every change

**Reply by bluebill1049** [reply]
yea, then why you need `useEffect`? 
```
const data = watch()
// why need useEffect here?
return {data};
```

**Reply by bluebill1049** [reply]
if you want data goes into the store each onChange, then it's easier to attach `onChange` on the actual input instead of watch

**Reply by Mohamedsobhy292** [reply]
`onChange`  does the work, thanks

**Reply by hk-skit** [reply]
@bluebill1049 Is there any API that can be used on the form for such cases, instead of manually attaching handler in onChange.
```
useForm({
handleChange // change handler.
});

```
